### PR TITLE
pcsctools: 1.6.0 -> 1.6.2

### PR DESCRIPTION
**Pull request by SebTM**
###### Description of changes
Update to latest release, Source-build, new Homepage

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Should homepage be updated as well?

**Comment by SebTM** [comment]
> Should homepage be updated as well?

You are right, also took a little more effort and made it a source-build instead of trusting the build-tars :)

File: pkgs/tools/security/pcsctools/default.nix [hunk 1]
```diff
@@ -13,20 +14,26 @@
...
+  '';
+
   postPatch = ''
     substituteInPlace ATR_analysis \
       --replace /usr/local/pcsc /etc/pcsc \
       --replace /usr/share/pcsc $out/share/pcsc
   '';
 
-  buildInputs = [ dbus perlPackages.perl pcsclite ]
+  buildInputs = [ autoreconfHook dbus perlPackages.perl pcsclite ]
```
Comment (NickCao) [inline pkgs/tools/security/pcsctools/default.nix:36]: `autoreconfHook` shall be in `nativeBuildInputs`

File: pkgs/tools/security/pcsctools/default.nix [hunk 2]
```diff
@@ -13,20 +14,26 @@
...
+    owner = "LudovicRousseau";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-c7md8m1llvz0EQqA0qY4aGb3guGFoj+8uS4hUTzie5o=";
   };
 
+  preConfigure = ''
+    ./bootstrap
+  '';
+
```
Comment (NickCao) [inline pkgs/tools/security/pcsctools/default.nix:29]: This is exactly what `autoreconfHook` does for you, no need to do it explicitly.

Comment (SebTM) [reply pkgs/tools/security/pcsctools/default.nix]: Thanks tired that and didn't worked I suspect because of the wrong buildInputs I wasn't aware of.

**Review by NickCao — APPROVED** [review]
diff LGTM, waiting for ofborg.

**Comment by NickCao** [comment]
@ofborg build pcsc-tools

**Comment by NickCao** [comment]
Oh the commit message should have been `pcsctools: 1.6.0 -> 1.6.2`

**Comment by NickCao** [comment]
@ofborg build pcsctools

**Comment by SebTM** [comment]
> Oh the commit message should have been `pcsctools: 1.6.0 -> 1.6.2`

fixed

**Pull request merged**